modules/apps/content-dashboard/content-dashboard-document-library-impl/src/main/java/com/liferay/content/dashboard/document/library/internal/item/action/PreviewFileEntryContentDashboardItemAction.java

```diff
@@ -63,26 +73,45 @@ public Type getType() {
 
 	@Override
 	public String getURL() {
-		InfoItemFieldValues infoItemFieldValues =
-			_infoItemFieldValuesProvider.getInfoItemFieldValues(_fileEntry);
-
-		return Optional.ofNullable(
-			infoItemFieldValues.getInfoFieldValue("previewURL")
-		).map(
-			InfoFieldValue::getValue
-		).orElse(
-			StringPool.BLANK
-		).toString();
+		PortletRequest portletRequest =
+			(PortletRequest)_httpServletRequest.getAttribute(
+				JavaConstants.JAVAX_PORTLET_REQUEST);
+
+		if (portletRequest == null) {
+			return null;
+		}
+
+		try {
+			String backURL = ParamUtil.getString(
+				_httpServletRequest, "backURL");
+
+			String portletNamespace = _portal.getPortletNamespace(
+				DLPortletKeys.DOCUMENT_LIBRARY_ADMIN);
+
+			return HttpComponentsUtil.addParameter(
```

Comment:
@cgoncas, maybe it would be better to use setParameter instead of addParameter? Just in the case that the URL already contains the parameter (which I don't know if it could happen in this case).

What do you think?

Reply:
Forget what I said :)